### PR TITLE
gene mutation status are rendered for GDC top mutated genes in geneset edit UI

### PR DESCRIPTION
**Pull request by airenzp**
## Description

Here it is a draft PR with the gene rendering. When using load top mutated genes the mutation information is provided and rendered. Please check

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [ ] Tests: added and/or passed unit and integration tests, or N/A
- [ ] Todos: commented or documented, or N/A
- [ ] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/dom/genesetEdit.ts [hunk 1]
```diff
@@ -181,7 +181,7 @@ export function showGenesetEdit(arg: showGenesetEditArg) {
 					const result = await vocabApi.getTopMutatedGenes(args)
 
 					geneList = []
-					for (const gene of result.genes) geneList.push({ name: gene.name })
+					for (const gene of result.genes) geneList.push({ name: gene.name, mutationStat: gene.mutationStat })
```
Comment (xzhou82) [inline client/dom/genesetEdit.ts:184]: can change to `geneList.push(...result.genes)`

File: client/dom/genesetEdit.ts [hunk 2]
```diff
@@ -300,6 +300,25 @@ export function showGenesetEdit(arg: showGenesetEditArg) {
 		if (hasChanged) submitBtn.node().focus()
 	}
 
+	function renderGene(gene) {
+		const div = select(this).style('background-color', '#fffdaf').style('border-radius', '5px')
+
+		div.insert('div').style('display', 'inline-block').html(gene.name)
+		if (gene.mutationStat) {
+			div.html(`${gene.name}&nbsp;&nbsp;`)
+			for (const m of gene.mutationStat) {
```
Comment (xzhou82) [inline client/dom/genesetEdit.ts:309]: m is either {class,count} or {dt,count}, please handle both types

File: client/dom/genesetEdit.ts [hunk 3]
```diff
@@ -300,6 +300,25 @@ export function showGenesetEdit(arg: showGenesetEditArg) {
...
+				const classinfo = mclass[m.class]
+				if (classinfo)
+					div
+						.insert('div')
+						.style('display', 'inline-block')
+						.style('background-color', classinfo.color)
+						.style('padding', '0px 2px')
+						.text(m.count)
```
Comment (xzhou82) [inline client/dom/genesetEdit.ts:317]: please use .8em font size

File: client/dom/genesetEdit.ts [hunk 4]
```diff
@@ -300,6 +300,25 @@ export function showGenesetEdit(arg: showGenesetEditArg) {
 		if (hasChanged) submitBtn.node().focus()
 	}
 
+	function renderGene(gene) {
+		const div = select(this).style('background-color', '#fffdaf').style('border-radius', '5px')
```
Comment (xzhou82) [inline client/dom/genesetEdit.ts:304]: please still use gray but not yellow for gene btn color

**Comment by xzhou82** [comment]
for gdc top mutated genes, the sample counts for cnv & ssm are now indicated, along with a color legend
it won't be rendered for custom genes or msigdb
it should not affect oncomatrix or hierclustering

**Review by congyu-lu — COMMENTED** [review]
using the GDC glioma TME genes link, load top genes for gene expression, and then load top genes for variables, see this error multiple times: 
getaddrinfo ENOTFOUND api.gdc.cancer.gov

**Comment by xzhou82** [comment]
i tried twice and it did fail once. i also saw this before with oncomatrix sporadically. so it's not due to these changes

**Pull request merged**